src/Azure/Orleans.Transactions.AzureStorage/TransactionalState/AzureTableTransactionalStateStorageFactory.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.WindowsAzure.Storage.Table;
+using Newtonsoft.Json;
+using Orleans.Configuration;
+using Orleans.Runtime;
+using Orleans.Serialization;
+using Orleans.Transactions.Abstractions;
+
+namespace Orleans.Transactions.AzureStorage.TransactionalState
+{
+    public class AzureTableTransactionalStateStorageFactory : ITransactionalStateStorageFactory, ILifecycleParticipant<ISiloLifecycle>
+    {
+        private readonly string name;
+        private readonly AzureTableTransactionalStateOptions options;
+        private readonly ClusterOptions clusterOptions;
+        private readonly ILoggerFactory loggerFactory;
+        private CloudTable table;
+
+        public AzureTableTransactionalStateStorageFactory(string name, AzureTableTransactionalStateOptions options, IOptions<ClusterOptions> clusterOptions, ILoggerFactory loggerFactory)
+        {
+            this.name = name;
+            this.options = options;
+            this.clusterOptions = clusterOptions.Value;
+            this.loggerFactory = loggerFactory;
+        }
+
+        public ITransactionalStateStorage<TState> Create<TState>(string stateName, IGrainActivationContext context) where TState : class, new()
+        {
+            string partitionKey = MakePartitionKey(context, stateName);
+            var settingsResolver = ActivatorUtilities.CreateInstance< JsonSerializerSettingsResolver>(context.ActivationServices);
```

Comment:
This feels a bit odd to me, since getting those settings creates a non-insignificant object graph. I feel that we should either:
* Use `IOptions<JsonSerializerSettings>` (which I realize might feel strange, but it works)
* Use `ActivatorUtilities.GetServiceOrCreateInstance` and optionally TryAdd `JsonSerializerSettings` to the container by default if this provider is being used
* Create the settings once in this class, therefore not creating them in the scope of the activation

I assume there's some intention that users can customize those settings on a per-activation basis, but I'm not sure that's worth the cost

Reply:
TL;RD
I suggest the following:
*  Resolving the json settings once and using the same setting for all the storage instances, even though that technically violates the grain scoping.
* Add issue to support pluggable state serialization.
* Add issue to design and formalize handling of json serialization settings management.

--

This sux, imo.  The problem is deeper than this PR so I'll open issues to address in future.

At a high level this is related to how we handle serialization for persistence, an unresolved issue.  I've opted to use json rather than the serialization manager as an initial starting point.

To support this I need to  call GetDefaultSerializerSettings, which requires the type resolver and grain factory from the container.  Since this is in the context of a grain these 'should' be resolved using the grains container, as it's scoped, which is the case.  I could resolve this once for all storage instances, but that technically violates the scoping.  We've done this before, so it's not unprecedented so I'll change that, but overall the larger issue still remains.

Reply:
It doesn't violate scoping. Violating scoping would be pulling something from a Scoped or Transient service from a Singleton or service. It's fine to pull a Singleton from a Scoped service.

If we move the JsonSettings up into the factory ctor then it's fine.

As for JSON vs SerializationManager, I'm not convinced that SerializationManager is inherently bad for persistence - it allows users to register serializers which are suitable for persistence and those will be used preferentially. The issue is that we want to somehow avoid using unsuitable serializers - almost as if having another SerializationManager which is configured without generated serializers or a fallback serializer.

Reply:
Regarding scoping.  Calling logic is not expected to know or care if the services are singleton, transient, scoped, or whatever, it only cares that it acquires a service from the container for its scope, which this would not, that's why I'm arguing it violates scoping.  Again, we've done this before, so I'm not necessarily opposed to it, but in general, I try to respect a grains scope as every violation risks unexpected behaviors in the future.

> As for JSON vs SerializationManager, I'm not convinced that SerializationManager is inherently bad for persistence.

Neither am I.  I claimed only that our persistence serialization problems are wider than this PR and unresolved.  Until we figure that out, serializing using json takes our serializers out of the picture.  Most users understand json, and understand the implications of it’s use, which is not the case for our serializers.